python/lsst/daf/butler/datastores/file_datastore/get.py

```diff
@@ -0,0 +1,538 @@
+                        # be expired whilst we are using it).
+                        can_be_cached = True
+
+                        # Say that it is "likely" to be cached because
+                        # if the formatter read fails we will not be
+                        # caching this file.
+                        cache_msg = " and likely cached"
+
+                    msg = f"(via download to local file{cache_msg})"
+
+                # Calculate the (possibly) new location for the formatter
+                # to use.
+                newLocation = Location(*local_uri.split()) if location_updated else None
+
+                log.debug(
+                    "Reading%s from location %s %s with formatter %s",
+                    f" component {getInfo.component}" if isComponent else "",
+                    uri,
+                    msg,
+                    formatter.name(),
+                )
+                try:
+                    with (
+                        formatter._updateLocation(newLocation),
+                        time_this(
+                            log,
+                            msg="Reading%s from location %s %s with formatter %s",
+                            args=(
+                                f" component {getInfo.component}" if isComponent else "",
+                                uri,
+                                msg,
+                                formatter.name(),
+                            ),
+                        ),
+                    ):
+                        result = formatter.read(component=getInfo.component if isComponent else None)
+                except Exception as e:
+                    raise ValueError(
+                        f"Failure from formatter '{formatter.name()}' for dataset {ref.id}"
+                        f" ({ref.datasetType.name} from {uri}): {e}"
+                    ) from e
+
+                # File was read successfully so can move to cache
+                if can_be_cached:
+                    cache_manager.move_to_cache(local_uri, cache_ref)
+
+    return post_process_get(
+        result, ref.datasetType.storageClass, getInfo.assemblerParams, isComponent=isComponent
+    )
+
+
+def get_dataset_as_python_object_from_get_info(
+    allGetInfo: list[DatastoreFileGetInformation],
+    *,
+    ref: DatasetRef,
+    parameters: Mapping[str, Any] | None,
+    cache_manager: AbstractDatastoreCacheManager,
+) -> Any:
+    """Retrieve an artifact from storage and return it as a Python object
+
+    Parameters
+    ----------
+    allGetInfo : `list`[`DatastoreFileGetInformation`]
+        Pre-processed information about each file associated with this artifact
+    ref : `DatasetRef`
+        The registry information associated with this artifact.
+    parameters : `Mapping`[`str`, `Any`]
+        `StorageClass` and `Formatter` parameters
+    cache_manager: `AbstractDatastoreCacheManager`
```

Comment:
```suggestion
    cache_manager : `AbstractDatastoreCacheManager`
```
hopefully some kind of linter would spot this. The numpydoc one says it will.